src/main/java/frc/robot/shuffleboard/ShuffleboardWidgets.java

```diff
@@ -157,40 +161,63 @@ public void periodic() {
   /**sets variables to data from the interfaces - gets called periodically*/
   private void ShuffleboardInformation() {
 
-    // Drivetrain Data INCOMPLETE
-    leftEncoderValue = drivetrain.getLeftEncoder();
-    rightEncoderValue = drivetrain.getRightEncoder();
-    // Gyro Datas COMPLETE
-    gyroAngleDegrees = drivetrain.getAngleDegrees();
-    Pose2d pose = drivetrain.getRobotPose();
-    robotX = pose.getTranslation().getX();
-    robotY = pose.getTranslation().getY();
-    robotRotation = pose.getRotation().getDegrees();
-
-    // Turret Data COMPLETE
-    turretDegrees = turret.getPosition() * (1 / Math.PI) * 180;
-    turretVelocity = turret.getVelocity() * (1 / Math.PI) * 180;
-
-    // Shooter Data COMPLETE
-    flywheelVelocity = shooter.getFlywheelSpeed() * (1 / Math.PI) * 180;
-    flywheelTemperature = shooter.getInternalTemperature();
-    hoodDegrees = shooter.getHoodAngle() * (1 / Math.PI) * 180;
-    hoodVelocity = shooter.getHoodVelocity() * (1 / Math.PI) * 180;
-
-    // Magazine Data COMPLETE
-    cellCount = magazine.getCellCount();
-
-    // Climbing Data COMPLETE
-    isBrakeset = lift.isBrakeSet();
-    isLiftFullyExtended = lift.isLiftFullyExtended();
-    isLiftFullyRetracted = lift.isLiftFullyRetracted();
-    liftExtension = lift.liftExtension();
-    isPinned = lift.isPinned();
-
-    isWinchEngaged = winch.isWinchEngaged();
+    // only pulls data from the drivetrain if it exists
+    if(drivetrain != null){
+      // Drivetrain Data INCOMPLETE
```

Comment:
The "Drivetrain Data" needs to be complete _(or at least labeled as such)_ before this is pushed.